float64[] action
---
float64[] next_state